### PR TITLE
`General`: Hide several buttons in course management when tutors should not see them 

### LINKED ISSUE #7467: Student user sees and can access Edit button in Competencies  tab
### Describe the bug

While testing on legacy ts2 I found out that the test-users 3 and 4 (role: Student) can see the "Edit" button in the Competencies tab and because of that can access the Management page. I tried to edit and delete competencies but I got an authorization error.

### To Reproduce

1. Go to a course with competencies 
2. Click on a competency
3. In the top right corner you see the "Edit" button
4. Click the button
5. You get an "You are not authorized to access this page." error
6. Click on "Competency Management"

Now you can try to access any page in the management view but are always going to get an authoritarian error. When you try to delete a competency you get a new error (see screenshot).

### Expected behavior

Not seeing the button and not being able to access the Edit and Management pages.

### Screenshots

image
image
image
image
image

### Which version of Artemis are you seeing the problem on?

6.6.3. TS2

### What browsers are you seeing the problem on?

Chrome

### Additional context

I talked with @JohannesStoehr and he checked the code. He said:
This bug is only for users who have the instructor role in another course.
It is because of `*jhiHasAnyAuthority` and therefore it is sufficient to have the respective role in any course.

Example: 
`*jhiHasAnyAuthority="['ROLE_ADMIN', 'ROLE_INSTRUCTOR']"`
Completely no matter where the user is an instructor, this is true and shows the component (for example).
Normally we use `*ngIf="course.isAtLeastInstructor"`

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).

### Motivation and Context

Fix #7467 and other similar problems related to `jhiHasAnyAuthority` leading to visible buttons that should not be visible.

### Description

`jhiHasAnyAuthority` only checks that the user has that authority in any course. So an instructor or tutor in one course could be a student in another but the directive would still show this html element to the user. Instead `isAtLeastTutor` and similar attributes of courses and exercises should be used in an `ngIf`.
If the whole route is already reserved for specific authorities, there is no use in checking them again inside the html.

The only visible change to the UI should be the button mentioned in the issue, the rest is removal and refactoring of redundant code.

### Steps for Testing

Prerequisites:
- 1 Instructor 
- 1 Student that is instructor in another course

1. Log in to Artemis
2. Create a competency as instructor
3. View the competency as the student that is instructor in another course
4. Check that there is no edit button on the right in the header

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by Hialus — DISMISSED** [review]
Code changes looks good to me. 👍🏻

**Review by JanaNF — APPROVED** [review]
Manual tested on legacy ts2. Worked as expected

**Review by MaximilianAnzinger — APPROVED** [review]
Code lgtm

File: src/main/webapp/app/course/competencies/competency.service.ts [hunk 1]
```diff
@@ -120,6 +122,9 @@ export class CompetencyService {
             res.body.exercises = ExerciseService.convertExercisesDateFromServer(res.body.exercises);
             res.body.exercises.forEach((exercise) => ExerciseService.parseExerciseCategories(exercise));
```
Comment (Strohgelaender) [inline src/main/webapp/app/course/competencies/competency.service.ts:123]: Can you also set the access rights for the exercises here for consitency?

**Review by sarpsahinalp — APPROVED** [review]
Tested on ts2. Works as expected!

**Review by egekurt123 — APPROVED** [review]
Works as expected on ts2

**Review by Strohgelaender — DISMISSED** [review]
code

**Review by Gusti2010 — APPROVED** [review]
tested on ts1 and works as expected!

**Review by JanaNF — DISMISSED** [review]
Manual tested on legacy ts2. Worked as expected

**Review by MaximilianAnzinger — DISMISSED** [review]
code

**Review by vinceclifford — CHANGES_REQUESTED** [review]
Tested on TS3. Tested with user nr. 4 as a student. Instructors were user nr. 13 & 14. However, the student was able to edit the competency. Tested on the course Vincent-Test-Course2

image
image

**Review by MaximilianAnzinger — APPROVED** [review]

**Review by vinceclifford — APPROVED** [review]
Reapproved. Tested on ts2

**Review by JanaNF — APPROVED** [review]
Tested again on ts2. Still works

**Review by Hialus — APPROVED** [review]
Reapprove code

**Pull request merged**